module/purpose/Soccer/src/Soccer.cpp

```diff
@@ -88,32 +105,53 @@ namespace module::purpose {
             emit<Task>(std::make_unique<FindPurpose>(), 1);
             // The robot should always try to recover from falling, if applicable, regardless of purpose
             emit<Task>(std::make_unique<FallRecovery>(), 2);
+            // Trigger Find purpose by emitting robocup
+            auto msg = std::make_unique<RoboCup>();
+            msg->current_pose.player_id = PLAYER_ID;
+            emit(msg);
         });
 
-        on<Provide<FindPurpose>>().then([this] {
+        on<Provide<FindPurpose>(), Trigger<RoboCup>>().then([this] (const RoboCup robocup) {
             // Make task based on configured purpose/soccer position
             switch (cfg.position) {
                 case Position::STRIKER: emit<Task>(std::make_unique<Striker>(cfg.force_playing)); break;
                 case Position::GOALIE: emit<Task>(std::make_unique<Goalie>(cfg.force_playing)); break;
                 case Position::DEFENDER: emit<Task>(std::make_unique<Defender>(cfg.force_playing)); break;
+                case Position::DYNAMIC:
+                    manage_active_robots(robocup);
+                    break;
                 default: log<NUClear::ERROR>("Invalid robot position");
             }
         });
 
         on<Trigger<Penalisation>>().then([this](const Penalisation& self_penalisation) {
+            // Need to remove penalised robot from active_robots
+            for (auto it = active_robots.begin(); it != active_robots.end(); ++it) {
+                if (it->robot_id == self_penalisation.robot_id) {
+                    active_robots.erase(it);
+                    break;
+                }
+            }
+
             // If the robot is penalised, its purpose doesn't matter anymore, it must stand still
             if (!cfg.force_playing && self_penalisation.context == GameEvents::Context::SELF) {
                 emit(std::make_unique<ResetWebotsServos>());
                 emit(std::make_unique<Stability>(Stability::UNKNOWN));
                 emit(std::make_unique<ResetFieldLocalisation>());
                 emit<Task>(std::unique_ptr<FindPurpose>(nullptr));
             }
+
+            give_directions();
         });
 
         on<Trigger<Unpenalisation>>().then([this](const Unpenalisation& self_unpenalisation) {
             // If the robot is unpenalised, stop standing still and find its purpose
             if (!cfg.force_playing && self_unpenalisation.context == GameEvents::Context::SELF) {
+                // TODO: check if we still need the FindPurpose thing
```

Comment:
yeah you'll want the FindPurpose Task to run as this robot is unpenalised and needs to find out what its position is now.

Reply:
oops forgot to ask about this. I think that emitting the Robocup msg should now trigger the FindPurpose code, so wondering if we need 

emit<Task>(std::make_unique<FindPurpose>(), 1);